### PR TITLE
*: move config file option query-log-max-len to sysvar

### PR DESCRIPTION
**Pull request by espresso98**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/33769

#### Problem Summary:
The option `query-log-max-len` has historically been a config option and `tidb-query-log-max-len` has existed as a session variable. Based on requirements from cloud & PM it should instead be a `GLOBAL` sysvar.
### What is changed and how it works?
This PR converts the option `query-log-max-len` to a global (cluster-wide) sysvar `tidb-query-log-max-len`, and removes the config option and the session sysvar.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
The TiDB configuration file option `query-log-max-len` has been replaced by the global system variable `tidb-query-log-max-len`.  This change makes it easier to modify this setting on a cluster wide basis.
```

### CODE REVIEW

**Comment by espresso98** [comment]
Please review @morgo

**Review by morgo — APPROVED** [review]

**Review by dveeden — APPROVED** [review]
```
sql> SET GLOBAL tidb_query_log_max_len=5;
Query OK, 0 rows affected (0.0047 sec)

sql> select sleep(id) from t1;
+-----------+
| sleep(id) |
+-----------+
|         0 |
+-----------+
1 row in set (5.0028 sec)
```

```
# Time: 2022-05-03T08:27:00.93426371+02:00
# Txn_start_ts: 432946339101278208
# User@Host: root[root] @ ::1 [::1]
# Conn_ID: 403
# Query_time: 5.002372906
# Parse_time: 0.000205323
# Compile_time: 0.000498304
# Rewrite_time: 0.000165694
# Optimize_time: 0.000127934
# Wait_TS: 0.000028787
# Cop_time: 0.000409313 Request_count: 1
# DB: test
# Is_internal: false
# Digest: 850922af8bcba539198bb9559d6e317af6768dbd53539fee0225f44600f68688
# Stats: t1:pseudo
# Num_cop_tasks: 1
# Cop_proc_avg: 0 Cop_proc_addr: store1
# Cop_wait_avg: 0 Cop_wait_addr: store1
# Mem_max: 539
# Prepared: false
# Plan_from_cache: false
# Plan_from_binding: false
# Has_more_results: false
# KV_total: 0
# PD_total: 0.000004407
# Backoff_total: 0
# Write_sql_response_total: 0.00000353
# Result_rows: 1
# Succ: true
# IsExplicitTxn: false
# Plan: tidb_decode_plan('hAPwkDAJM18zCTAJMQlzbGVlcChjYXN0KHRlc3QudDEuaWQsIGRvdWJsZSBCSU5BUlkpKS0+Q29sdW1uIzIJMQl0aW1lOjVzLCBsb29wczoyLCBDb25jdXJyZW5jeTpPRkYJMzgwIEJ5dGVzCU4vQQoxCTMxXzUJMAkxCWRhdGE6VGFibGVGdWxsU2Nhbl80CTEJdGkBUhgxOC4xwrVzHVhwY29wX3Rhc2s6IHtudW06IDEsIG1heDogMjc5LjkFK0Rwcm9jX2tleXM6IDAsIHJwY18RKQEMBaoUIDE5OC40BS54Y29wcl9jYWNoZV9oaXRfcmF0aW86IDAuMDB9CTE1OR20JDIJNDNfNAkxXzABnwGxiDp0MSwga2VlcCBvcmRlcjpmYWxzZSwgc3RhdHM6cHNldWRvASsIaWt2CbMAewV9EDE1My42BXwAbCUtKDB9CU4vQQlOL0EK')
# Plan_digest: e81eea3563158d75f62a9ce11cffdbdb995e1c92323448dbebc16f8f154e2c46
"selec"(len:24);
```

File: config/config.go [hunk 1]
```diff
@@ -931,6 +929,7 @@ var deprecatedConfig = map[string]struct{}{
 	"stmt-summary.refresh-interval":      {},
 	"stmt-summary.history-size":          {},
 	"mem-quota-query":                    {},
+	"query-log-max-len":                  {},
```
Comment (bb7133) [inline config/config.go:932]: It should be 'log.query-log-max-len', see #34771

**Pull request merged**